src/utils/telnet-client.ts

```diff
@@ -107,9 +107,7 @@ export class TelnetClient {
     oldOne.removeListener('connect', this.onConnect);
     oldOne.removeListener('error', this.onError);
 
-    oldOne.end(() => {
-      oldOne.destroy().unref();
-    });
+    oldOne.destroy().unref();
```

Comment:
The solution could be nice, but googling into the difference between end and destroy, I've found a memory leak issue when you don't use end(). Perhaps, in your case of non-existent ip it would be better to catch an error instead